packages/api/amplify_api_dart/lib/src/graphql/web_socket/blocs/web_socket_bloc.dart

```diff
@@ -507,6 +528,41 @@ class WebSocketBloc with AWSDebuggable, AmplifyLoggerMixin {
     );
   }
 
+  /// Process life cycle stream monitors when the process resumes from a paused state.
+  StreamSubscription<ProcessStatus> _getProcessLifecycleStream() {
+    var prev = ProcessStatus.detached;
+    return _processLifeCycle.onStateChanged.listen(
+      (state) {
+        if (_isUnpausing(state, prev)) {
+          // ignore: invalid_use_of_internal_member
+          if (!WebSocketOptions.autoReconnect) {
+            _shutdownWithException(
+              const NetworkException(
+                'Unable to recover network connection, web socket will close.',
+                recoverySuggestion: 'Avoid pausing the process.',
+              ),
+              StackTrace.current,
+            );
+          } else {
+            add(const ProcessResumeEvent());
+          }
+        }
+
+        prev = state;
+      },
+      onError: (Object e, StackTrace st) =>
+          logger.error('Error in process life cycle stream $e, $st'),
+    );
+  }
+
+  bool _isUnpausing(ProcessStatus current, ProcessStatus previous) {
+    if (previous != ProcessStatus.paused) return false;
+
+    return current == ProcessStatus.hidden ||
+        current == ProcessStatus.inactive ||
+        current == ProcessStatus.resumed;
```

Comment:
Q: Naive question, is the process unpausing when status is not resumed?

Reply:
Technically it should always go from paused -> hidden -> inactive -> resumed per [Flutter documentation](https://api.flutter.dev/flutter/widgets/AppLifecycleListener-class.html) I just included inactive and resumed in the check as a fail safe. With that said onRestart might be the better terminology to use rather than unpaused or Resume.

Hidden, inactive, and resumed are all acceptable states for reconnecting the Websocket, the only difference between them is if the UI is being displayed and if a control is selected.

Reply:
Ah thanks for the context, that makes more sense. You can decide on the name, either `resume` or `onRestart` would work well in my opinion.